src/Controller/Controller.php

```diff
@@ -572,6 +572,9 @@ public function render($view = null, $layout = null)
             return $this->response;
         }
 
+        // Re-fetch View class to pass view variables set in beforeRender callbacks
+        $this->View = $this->getView();
+
         if ($viewClass !== $this->viewClass) {
             $this->View = $this->getView($this->viewClass);
```

Comment:
View vars would be passed even when recreating new view instance, just not the other properties.

Using a proxy object would be the way to go to totally avoid this issue.